odl/solvers/functional/default_functionals.py

```diff
@@ -0,0 +1,731 @@
+
+        Parameters
+        ----------
+        x : `LinearSpaceVector`
+            Element in the domain of the functional.
+
+        Returns
+        -------
+        out : `field` element
+            Result of the evaluation.
+        """
+        return np.abs(x).inner(self.domain.one())
+
+    @property
+    def gradient(x):
+        """Gradient operator of the functional.
+
+        Notes
+        -----
+        The operator that corresponds to the mapping
+
+        .. math::
+
+            x \\to \\nabla f(x)
+
+        where :math:`\\nabla f(x)` is the element used to evaluate
+        derivatives in a direction :math:`d` by
+        :math:`\\langle \\nabla f(x), d \\rangle`.
+        """
+        # It does not exist on all of the space
+        raise NotImplementedError('functional not differential.')
+
+    def proximal(self, sigma=1.0):
+        """Return the proximal operator of the functional.
+
+        Parameters
+        ----------
+        sigma : positive float, optional
+            Step size-like parameter
+
+        Returns
+        -------
+        out : Operator
+            Domain and range equal to domain of functional
+        """
+
+        return proximal_l1(space=self.domain)(sigma)
+
+    @property
+    def conjugate_functional(self):
+        """The convex conjugate functional of the L1-norm."""
+        return IndicatorLinftyUnitBall(self.domain)
+
+
+class IndicatorLinftyUnitBall(Functional):
+
+    """The indicator function on the unit ball, in the L-infinity norm.
+
+    This functional is defined as
+
+        ``f(x) = 0 if ||x||_infty <= 1, infty else,``
+
+    where ``||x||_infty`` is the infinity norm
+
+        ``||x||_infty = max(|x|)``.
+
+    Notes
+    -----
+    This functional is the convex conjugate functional of the L1-norm.
+    """
+
+    def __init__(self, domain):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        domain : `LinearSpace`
+            Domain of the functional.
+        """
+        super().__init__(domain=domain, linear=False)
+
+    def _call(self, x):
+        """Applies the functional the given point.
+
+        Parameters
+        ----------
+        x : `LinearSpaceVector`
+            Element in the domain of the functional.
+
+        Returns
+        -------
+        out : `field` element
+            Result of the evaluation.
+        """
+        if np.max(np.abs(x)) > 1:
+            return np.inf
+        else:
+            return 0
+
+    @property
+    def gradient(x):
+        """Gradient operator of the functional.
+
+        Notes
+        -----
+        The operator that corresponds to the mapping
+
+        .. math::
+
+            x \\to \\nabla f(x)
+
+        where :math:`\\nabla f(x)` is the element used to evaluate
+        derivatives in a direction :math:`d` by
+        :math:`\\langle \\nabla f(x), d \\rangle`.
+        """
+        # It does not exist on all of the space, only for ||x|| < 1
```

Comment:
It's good that you want to inform the user about the reason why there is no gradient. I think that information can go into the class docstring, and the error can be raised by the parent.